client/src/components/App/Content/LandingPage/adminLandingPage/TimeRangeFilterCard/useTimeRangeFilterCard.ts

```diff
@@ -1,28 +1,59 @@
 import { useState } from 'react';
+import { format } from 'date-fns';
 
 import { TimeRange } from 'models/TimeRange';
 import timeRanges from 'models/enums/timeRanges';
 
 import AdminLandingPageFilters from '../AdminLandingPageFilters';
 
+const dateMissingError = 'יש לבחור תאריך תחילה וסיום';
+const rangeError = 'טווח תאריכים צריך להיות תקין';
+
 const useTimeRangeFilterCard = (props : Props) => {
     
+    const {timeRangeFilter , setTimeRangeFilter, investigationInfoFilter, setInvestigationInfoFilter} = props;
     const defaultTimeRange = timeRanges[0];
+    const customTimeRange = timeRanges[4];
     const [isLoading, setIsLoading] = useState<boolean>(false);
-
-    const {timeRangeFilter , setTimeRangeFilter, investigationInfoFilter, setInvestigationInfoFilter} = props;
+    const [errorMes, setErrorMes] = useState<string>('');
 
     const onTimeRangeChange = (timeRangeId: number) => {
         const selectedTimeRange = timeRanges.find(timeRange => timeRange.id === timeRangeId) as TimeRange;
         setTimeRangeFilter(selectedTimeRange)
     }
 
+    const onStartDateSelect = (startDateInput: Date) => {
+        setTimeRangeFilter((timeRangeFilter) => {
+            return {...timeRangeFilter, startDate: format(startDateInput,'yyyy-MM-dd')};
+        });
+    }
+
+    const onEndDateSelect = (endDateInput :Date) => {
+        setTimeRangeFilter((timeRangeFilter) => {
+            return {...timeRangeFilter, endDate: format(endDateInput,'yyyy-MM-dd')};
+        });        
+    }
+
     const onUpdateButtonCLicked = () => {
         if(timeRangeFilter.id !== defaultTimeRange.id) {
-            setInvestigationInfoFilter({
-                ...investigationInfoFilter,
-                timeRange : {startDate: timeRangeFilter.startDate, endDate: timeRangeFilter.endDate}
-            })
+            if (timeRangeFilter.id == customTimeRange.id) {
+                if (timeRangeFilter.startDate === null || timeRangeFilter.endDate === null) {
```

Comment:
can timeRangeFilter be undefined? if so I think that Boolean(timeRangeFilter) will be more appropriate

Reply:
its because when someone chose "בחירת תאריכים", he can leave the startDate/endDate null so I'm checking that

Reply:
cool